libraries/chain/controller.cpp

```diff
@@ -1606,7 +1620,6 @@ struct controller_impl {
                 emit(self.applied_transaction, std::tie(trace, trx->packed_trx()));
             }
 
-
             if ( trx->is_transient() ) {
                // remove trx from pending block by not canceling 'restore'
                trx_context.undo(); // this will happen automatically in destructor, but make it more explicit
```

Comment:
I think the comment is wrong, I don't see `undo()` being called in `transaction_context`'s destructor.

Reply:
`undo_session` is a std::optional<session>` which will be destroyed. It is the `session`'s destructor that does the `undo()`.